base/cvd/cuttlefish/common/libs/fs/shared_fd.cpp

```diff
@@ -524,6 +527,25 @@ SharedFD SharedFD::Mkstemp(std::string* path) {
   }
 }
 
+Result<std::pair<SharedFD, std::string>> SharedFD::Mkostemp(
+    const std::string& path) {
+  // mkostemp replaces the Xs with random selections to make a unique filename
+  auto temp_path = path + "XXXXXX";
+  const int fd = mkostemp(temp_path.data(), O_CLOEXEC);
+  CF_EXPECTF(fd != -1, "Error creating temporary file: {}", strerror(errno));
+  auto shared_fd =
+      SharedFD(std::shared_ptr<FileInstance>(new FileInstance(fd, 0)));
+  CF_EXPECTF(shared_fd->IsOpen(),
+             "Error creating SharedFD from existing file descriptor for \"{}\"",
+             path);
+  CF_EXPECTF(close(fd) != -1,
+             "Error closing original file descriptor after creating SharedFD "
+             "for \"{}\": ",
+             path, strerror(errno));
```

Comment:
This will actually close the file descriptor before returning it, it was only needed because of the `Dup`. By passing the fd to the SharedFD constructor the newly created object takes ownership of that fd and you don't need to worry about it anymore.

Reply:
Thanks for the catch!